$NetBSD: patch-ba,v 1.1 2007/11/01 13:08:05 rillig Exp $

The array size of a local variable must be a constant expression, says
sunpro.

--- src/Ewmh.cc.orig	2007-03-19 20:57:09.000000000 +0100
+++ src/Ewmh.cc	2007-10-31 01:07:03.044178000 +0100
@@ -454,7 +454,7 @@ void Ewmh::updateWorkspaceNames(BScreen 
     const BScreen::WorkspaceNames &workspacenames = screen.getWorkspaceNames();
     const size_t number_of_desks = workspacenames.size();
 
-    char *names[number_of_desks];
+    char **names = new char *[number_of_desks];
 
     for (size_t i = 0; i < number_of_desks; i++) {
         names[i] = new char[workspacenames[i].size() + 1]; // +1 for \0
@@ -483,6 +483,7 @@ void Ewmh::updateWorkspaceNames(BScreen 
 
     for (size_t i = 0; i < number_of_desks; i++)
         delete [] names[i];
+    delete[] names;
 
 }
 
